### PR TITLE
feat(DLI): add new resource to manage global variable

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccGlobalVariable'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccGlobalVariable -timeout 360m -parallel 4
=== RUN   TestAccGlobalVariable_basic
=== PAUSE TestAccGlobalVariable_basic
=== CONT  TestAccGlobalVariable_basic
--- PASS: TestAccGlobalVariable_basic (17.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       17.436s
```

### CODE REVIEW

File: docs/data-sources/iec_keypair.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ data "huaweicloud_iec_keypair" "kp_1" {
 The following arguments are supported:
 
 * `name` - (Required, String) Specifies a unique name for the keypair. This parameter can contain a maximum of 64
-  characters, which may consist of letters, digits, underscores (_), and hyphens (-).
+  characters, which may consist of letters, digits, underscores (_), and hyphens (-).
```
Comment (ShiChangkuo) [inline docs/data-sources/iec_keypair.md]: please revert this file

File: docs/resources/dli_global_variable.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# huaweicloud_dli_global_variable
+
+Manages a DLI global variable resource within HuaweiCloud.  
+
+## Example Usage
+
+```hcl
+  resource "huaweicloud_dli_global_variable" "test" {
+    name         = "demo"
+    value        = "123"
```
Comment (ShiChangkuo) [inline docs/resources/dli_global_variable.md]: format error

File: docs/resources/dli_global_variable.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  Changing this parameter will create a new resource.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID which equals the name.
+
+* `project_id` - Project ID.
```
Comment (ShiChangkuo) [inline docs/resources/dli_global_variable.md]: it's better to delete this attribute

File: huaweicloud/services/dli/resource_huaweicloud_dli_global_variable.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	createGlobalVariableOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createGlobalVariableOpt.JSONBody = utils.RemoveNil(buildCreateGlobalVariableBodyParams(d))
+	_, err = createGlobalVariableClient.Request("POST", createGlobalVariablePath, &createGlobalVariableOpt)
+	if err != nil {
+		return diag.Errorf("error creating GlobalVariable: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/resource_huaweicloud_dli_global_variable.go]:
```suggestion
		return diag.Errorf("error creating DLI global variable: %s", err)
```
please update it in other places.

File: docs/data-sources/iec_keypair.md (hunk 1 above)
Comment (hmbig2) [reply docs/data-sources/iec_keypair.md]: done

File: docs/resources/dli_global_variable.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/dli_global_variable.md]: done

File: huaweicloud/services/dli/resource_huaweicloud_dli_global_variable.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/dli/resource_huaweicloud_dli_global_variable.go]: done, and will fix it in generator

**Pull request merged**